### PR TITLE
feat(whale-api): add GET api for on-chain governance

### PR DESCRIPTION
**Pull request by kyleleow**
#### What this PR does / why we need it:
- Add `/proposals`, `/proposals/:id` and `/proposals/:id/votes` endpoints on whale
- To allow straightforward paginated listing implementation for downstream, e.g. scan and wallet

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
- Intermittent codecov issue when running CI, seems to be related to https://github.com/codecov/codecov-action/issues/598

### CODE REVIEW

File: apps/whale-api/src/module.api/governance.controller.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+      {
+        proposalId: cfpProposalId,
+        masternodeId: masternodeId,
+        cycle: 1,
+        vote: ProposalVoteResultType.NEUTRAL
+      }
+    ])
+  })
+
+  it.only('should listProposalVotes with all records when limit is 0', async () => {
```
Comment (eli-lim) [inline apps/whale-api/src/module.api/governance.controller.e2e.ts]: Unintended change? I believe this skips the other tests

Comment (kyleleow) [reply apps/whale-api/src/module.api/governance.controller.e2e.ts]: 🙏 found broken tests after updating, fixing

Comment (kyleleow) [reply apps/whale-api/src/module.api/governance.controller.e2e.ts]: @eli-lim for some reason my recent push is not reflecting on this PR, possibly due to my access changes after i accepted the invite into jellyfish org, do you mind checking the new one  here: https://github.com/JellyfishSDK/jellyfish/pull/1970 , thanks!

**Comment by kyleleow** [comment]
Closed due to duplicate of https://github.com/JellyfishSDK/jellyfish/pull/1970

**Pull request closed without merging**